### DISCUSSION TITLE
Styles as themes (instead of resources)

### DISCUSSION THREAD

**Discussion by bsdRuf** [AvaloniaUI/Avalonia#13052] [Q&A] [answered]
Is it possible to have a LightStyle and a DarkStyle instead of using "Resources" as themes?
What would be the best way to let the user choose the style and remember it on the next start?

Background to my question: I am using treeDataGrid and DataGrids on some views and declairing Textblocks with style-classes instead of overwriting the default does not effect the treeDataGrid and DataGrids.

**Comment by thevortexcloud** [comment]
If you just want to change theme variants at run time, check the ControlCataloge example. It lets you change theme variants on the fly.

If you want to define custom theme variants there is a whole API for that.

https://docs.avaloniaui.net/docs/next/guides/styles-and-resources/how-to-use-theme-variants


If you want it to persist you can probably just load the saved value from somewhere. But it's hard to say for sure without seeing code what exactly you are trying to do.

**Comment by bsdRuf** [comment]
Thanks for the quick reply. Sorry for being unclear, with what I'm trying to do. In a folder Styles I got BSDStyles_dark and BSDStyles_light

I am trying to use Window.Styles as Themes. But using Binding at the StyleInclude isnt working:
Avalonia.Markup.Xaml.XamlLoadException: "No precompiled XAML found for /Style/BSDStyles_dark.axaml (baseUri: avares://BSD_Avalonia/App.axaml), make sure to specify x:Class and include your XAML file as AvaloniaResource"

Atm, I need to manually switch between dark and light Theme in the App.axaml

```
<Application xmlns="https://github.com/avaloniaui"
             xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
             x:Class="BSD_Avalonia.App">
	<Application.Styles>
		<FluentTheme/>
		<StyleInclude Source="avares://Avalonia.Controls.TreeDataGrid/Themes/Fluent.axaml"/>
		<StyleInclude Source="avares://Avalonia.Controls.DataGrid/Themes/Fluent.xaml"/>
		<StyleInclude Source="avares://Avalonia.Controls.ColorPicker/Themes/Fluent/Fluent.xaml" />
		
		<StyleInclude Source="/Style/BSDStyles_dark.axaml"/>
		<!--<StyleInclude Source="/Style/BSDStyles_light.axaml"/>-->
	</Application.Styles>
</Application>
```

As far as I understood am just able to set <Style.Resources> as AvaloniaResource by x:Key='Light' and x:Key='Dark'

Code sample of what I'm trying to do:

CurrentStyle is set in the MainViewModel to "/Style/BSDStyles_dark.axaml" (which is the correct path, at least it was working when StyleInclude was static).

```
<Window xmlns="https://github.com/avaloniaui"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        mc:Ignorable="d" d:DesignWidth="1100" d:DesignHeight="700"
        xmlns:m="using:BSD_Avalonia.Treedatagrid"
		xmlns:vm="using:BSD_Avalonia.ViewModels"
		x:DataType="vm:MainViewModel"
		WindowStartupLocation="CenterScreen"
		xmlns:local="using:BSD_Avalonia"
		xmlns:lvc="using:LiveChartsCore.SkiaSharpView.Avalonia"
        x:Class="BSD_Avalonia.MainWindow"
        Title="BSD_Avalonia">
	<Window.Styles>
		<!--Instead of: <StyleInclude Source="/Style/BSDStyles_dark.axaml"/>-->
		<StyleInclude Source="[Binding CurrentStyle}"/>
	</Window.Styles>
        ....

```

**Reply by thevortexcloud** [reply, marked as answer]
I am not sure if you can use a binding like this (although it may work if you change it to a URI [using "avares://" with the full assembly path](https://docs.avaloniaui.net/docs/getting-started/assets)) instead of a regular path. You could use the runtime loader to parse the style file and set the window style in the code behind.

**Reply by bsdRuf** [reply]
By using the assembly path I got the same error but the idea was worth the try. The runtime loader worked like I wanted it to. Thanks!

**Reply by thevortexcloud** [reply]
You are welcome!

**Comment by kyuranger** [comment]
```
 var S=AvaloniaXamlLoader.Load(new Uri("avares://YourAssembly/Styles/DialogHost.axaml")) as Styles;
 if (S is { })
 {
     App.Current.Styles.Add(S);
 }
```

**Reply by thevortexcloud** [reply]
Note that `resm` is apparently deprecated.

https://docs.avaloniaui.net/docs/getting-started/assets

**Reply by kyuranger** [reply]
Yes, you are right. And now I have found a new way to solve this. Please check my new answer.

**Reply by bsdRuf** [reply]
It was also a working solution. Thanks for the quick answers!
